Add TRAMP support

wedens opened the pull request:
Make it work via TRAMP connection when atool is available on remote (or locally
when used with `/sudo:` method).

Tested locally and with TRAMP ssh and sudo methods.

wedens committed:
Add TRAMP support.

Make it work via TRAMP connection when atool is available on remote (or locally
when used with `/sudo:` method).

HKey commented:
Hi.

Thank you for the PR.

I tested your branch on emacs 27.0.50.

It works in almost cases, but when the output directory doesn't exist yet on a TRAMP environment, dired-atool makes the needed directory on the local environment, then atool command fails.
The expected vihavior is making the directory on the TRAMP environment.
dired-atool makes a directory via `make-directory` if needed when it doesn't exist.
So, I think that `dir` local variable has to keep a TRAMP path format (e.g. "/sudo:localhost:/tmp") to make the directory on the TRAMP environment.  
Then convert `dir` to a local path format (e.g. "/tmp") before passing it to `command-list` local variable.  

About converting file paths returned by `(dired-get-marked-files t arg)`, is that necessary for handling any special cases?  
When I tested, the value returned by `(dired-get-marked-files t arg)` already seemed local path formats.  
If there are special cases, will you tell me that?

wedens commented:
> About converting file paths returned by (dired-get-marked-files t arg), is that necessary for handling any special cases?

I've tested it again and it indeed always returns local names. I don't remember why I added additional path conversion, to be honest. I got bitten by it multiple times, so perhaps I added it just to be safe.


I've removed this (probably) unnecessary check and fixed directory creation.

HKey commented:
Thank you, the directory creation works!

> I don't remember why I added additional path conversion, to be honest. I got bitten by it multiple times, so perhaps I added it just to be safe.

I got it. I'll leave it not converted until any situation is found that the conversion is needed.
